dali/test/python/test_dltensor_operator.py

```diff
@@ -0,0 +1,165 @@
+import nvidia.dali.ops as ops
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.types as types
+import torch.utils.dlpack as torch_dlpack
+import torch
+import os
+import random
+import numpy
+from mxnet import ndarray as mxnd
+
+test_data_root = os.environ['DALI_EXTRA_PATH']
+images_dir = os.path.join(test_data_root, 'db', 'single', 'jpeg')
+
+
+def random_seed():
+    return int(random.random() * (1 << 32))
+
+
+DEVICE_ID = 0
+BATCH_SIZE = 8
+ITERS = 32
+SEED = random_seed()
+NUM_WORKERS = 6
+
+
+class CommonPipeline(Pipeline):
+    def __init__(self):
+        super(CommonPipeline, self).__init__(BATCH_SIZE, NUM_WORKERS, DEVICE_ID, seed=SEED,
+                                             exec_async=False, exec_pipelined=False)
+        self.input = ops.FileReader(file_root=images_dir)
+        self.decode = ops.ImageDecoder(device='cpu', output_type=types.RGB)
+        self.resize = ops.Resize(resize_x=3, resize_y=3)
+
+    def load(self):
+        jpegs, labels = self.input()
+        decoded = self.decode(jpegs)
+        return self.resize(decoded)
+
+
+class LoadingPipeline(CommonPipeline):
+    def __init__(self):
+        super(LoadingPipeline, self).__init__()
+
+    def define_graph(self):
+        im = self.load()
+        im2 = self.load()
+        return im, im2
+
+
+class DLTensorOpPipeline(CommonPipeline):
+    def __init__(self, function):
+        super(DLTensorOpPipeline, self).__init__()
+        self.op = ops.DLTensorPythonFunction(function=function, num_outputs=2)
+
+    def define_graph(self):
+        im = self.load()
+        im2 = self.load()
+        return self.op(im, im2)
+
+
+def torch_adapter(fun, in1, in2):
+    tin1 = [torch_dlpack.from_dlpack(dltensor) for dltensor in in1]
+    tin2 = [torch_dlpack.from_dlpack(dltensor) for dltensor in in2]
+    tout1, tout2 = fun(tin1, tin2)
+    return [torch_dlpack.to_dlpack(tout) for tout in tout1], \
+           [torch_dlpack.to_dlpack(tout) for tout in tout2]
+
+
+def torch_wrapper(fun):
+    return lambda in1, in2: torch_adapter(fun, in1, in2)
+
+
+def torch_case(fun):
+    load_pipe = LoadingPipeline()
+    op_pipe = DLTensorOpPipeline(function=torch_wrapper(fun))
+
+    load_pipe.build()
+    op_pipe.build()
+
+    for iter in range(ITERS):
```

Comment:
Unless there is reason not to, maybe use `compare_pipelines` from `test_utils`?

Reply:
`compare_pipelines` does not give any option to postprocess pipeline output, so I could add PythonFunction at the end of the reference pipeline, but it will not be possible when I  move on to a GPU operator, so I leave it as it is